### PR TITLE
feat: add ws query params

### LINKED ISSUE #526: Add query params in Websockets

### PR DESCRIPTION
**Pull request by sansyrox**
**Description**

This PR fixes #526

### CODE REVIEW

File: src/websockets/mod.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ pub struct WebSocketConnector {
     pub router: HashMap<String, FunctionInfo>,
     pub task_locals: TaskLocals,
     pub registry_addr: Addr<WebSocketRegistry>,
+    pub query_params: HashMap<String, String>,
```
Comment (sansyrox) [inline src/websockets/mod.rs:29]: use the QueryParams datastructure

File: src/websockets/mod.rs [hunk 2]
```diff
@@ -206,13 +212,17 @@ pub async fn start_web_socket(
...
     let registry_addr = get_or_init_registry_for_endpoint(endpoint);
+    let query_params = req.query_string();
 
     ws::start(
         WebSocketConnector {
             router,
             task_locals,
             id: Uuid::new_v4(),
             registry_addr,
+            query_params: web::Query::<HashMap<String, String>>::from_query(query_params)
```
Comment (sansyrox) [inline src/websockets/mod.rs:223]: same here

**Comment by sansyrox** [comment]
Add documentation

**Comment by sansyrox** [comment]
Add integration tests

**Comment by VishnuSanal** [comment]
superseded by #886

**Comment by sansyrox** [comment]
Thanks @VishnuSanal

**Pull request closed without merging**